src/lib/cpd/CpdWrapper.ts

```diff
@@ -0,0 +1,98 @@
+import {Logger} from '@salesforce/core';
+import {FileHandler} from '../util/FileHandler';
+import {ChildProcessWithoutNullStreams} from 'child_process';
+import { PmdSupport} from '../pmd/PmdSupport';
+import * as JreSetupManager from './../JreSetupManager';
+import path = require('path');
+
+const MAIN_CLASS = 'net.sourceforge.pmd.cpd.CPD';
+const HEAP_SIZE = '-Xmx1024m';
+
+interface CpdWrapperOptions {
+	path: string;
+	language: string;
+	minimumTokens: number;
+}
+
+export default class CpdWrapper extends PmdSupport {
```

Comment:
The cpd engine uses PmdSupport so that it can use the methods already implemented for pmd.